spec/integration/user_post_index_page

```diff
@@ -0,0 +1,23 @@
+require 'rails_helper'
+
+RSpec.describe 'User Post Index Page', type: :feature do
+  let!(:user) { create(:user_with_comments_and_likes) }
+
+  it 'displays user and post information correctly' do
+    visit user_posts_path(user)
+
+    expect(page).to have_content(user.username)
+    expect(page).to have_selector("img[src$='#{user.profile_picture_url}']")
+    expect(page).to have_content("Posts: #{user.posts.count}")
+
+    user.posts.each do |post|
+      expect(page).to have_content(post.title)
+      expect(page).to have_content(post.body[0..100])
+      expect(page).to have_content("Comments: #{post.comments.count}")
+      expect(page).to have_content("Likes: #{post.likes.count}")
+    end
+
+    click_link user.posts.first.title
+    expect(current_path).to eq(post_path(user.posts.first))
+  end
```

Comment:
- [ ] Same things for this file too. Seme tests are missing. Kindly check the screenshot to know what are those missing tests.


### Passed Tests
![test-passed](https://github.com/Ashnaali3255/Blog-website/assets/26597432/f3288932-af9f-4324-8385-d4fa8b8d8dbb)


### Missing Tests
![user-post-index-page](https://github.com/Ashnaali3255/Blog-website/assets/26597432/f7bb6403-5078-44a6-9829-ac0d3e0325e1)